tools/js-tools/jest/jest-resolver.js

```diff
@@ -8,6 +6,28 @@ const badBrowserPackages = new Set( [
 	'parsel-js',
 ] );
 
+// TypeScript looks for various different extensions on imports. We need to reproduce this here.
+// Based on code from https://github.com/VitorLuizC/ts-jest-resolver/blob/98fdae29c6622ae67f2e10ecc03af7f79d16d24d/src/index.ts
+// (which doesn't seem to handle tsc's --jsx option)
+const resolutions = [
```

Comment:
One difference I notice between `ts-jest-resolver` and this is that the original script doesn't have the matcher in the list of extensions. It seems intentional, as we're throwing an error when the extensions match, though I'm not sure why.

Reply:
The logic in ts-jest-resolver is "try these extensions first, then fall back to what was requested". But I found in local testing that for `.js` it seems TypeScript looks for `.ts`, `.tsx`, `.js`, and `.jsx` in that order, while the fallbacking logic would have to do `.js` last. So I adjusted it to include `.js` in the list, and save the error if it fails so we can re-throw it after checking for `.jsx` (rather than having it check for `.js` twice).